### ISSUE TITLE
Infinite loop when "Addmusic_sample groups.txt" includes invalid data

### ISSUE THREAD

**Issue by spooonsss** [KungFuFurby/AddMusicKFF#395]
I assumed `Addmusic_sample groups.txt` would support comments, but it puts the parser into an infinite loop instead.

Try Addmusic_sample groups.txt:
```

;f
```

https://github.com/KungFuFurby/AddMusicKFF/blob/312f3454b87808baac4001e762f634e89cac2280/src/AddmusicK/AddmusicK.cpp#L609 needs an `else` that throws an error.

(I'd suggest it support comments too)

**Comment by KungFuFurby** [comment]
Yup. I can see how this can occur: simply by not even finding a sample group name to begin with and not having gotten any samples either and running into any non-alphanumeric character. For whatever reason, I may have other suspicions on the ones that do output errors...

**Comment by jneen** [comment]
Ran into this today again. Many sample group files in the archive contain comments beginning with `;` - it would be nice if comments like these were supported (or at the very least if the program exited reasonably with an error).

**Comment by jneen** [comment]
Example: https://www.smwcentral.net/?p=section&a=details&id=11467

Note the several comments there saying the port "doesn't work with AddmusicK", likely because the comment was copied along with the sample group.

**Comment by KungFuFurby** [comment]
Acknowledged. I have commented within your example case.
